src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -49,41 +49,46 @@ namespace System
         }
 
         // Creates a new guid from a read-only span.
-        public Guid(ReadOnlySpan<byte> b) : this(b, false) { }
-
-        public Guid(ReadOnlySpan<byte> b, bool bigEndian)
+        public Guid(ReadOnlySpan<byte> b)
         {
             if (b.Length != 16)
             {
-                ThrowArgumentException();
+                ThrowGuidArrayCtorArgumentException();
+            }
+
+            this = Unsafe.As<byte, Guid>(ref MemoryMarshal.GetReference(b));
```

Comment:
This should use unaligned read like the original code.

Was there anything wrong with the original code? The JIT should eliminate the bounds check just fine.